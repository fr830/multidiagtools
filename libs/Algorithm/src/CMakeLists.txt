

find_package(Qt5 COMPONENTS Core)

# set(CMAKE_INCLUDE_CURRENT_DIR ON)

set(SOURCE_FILES
  Mdt/Algorithm.cpp
)

mdt_add_library(
  NAME Algorithm
  SOURCE_FILES ${SOURCE_FILES}
  HEADERS_DIRECTORY .
  LINK_DEPENDENCIES Qt5::Core
)
mdt_set_library_description(
  NAME Algorithm
  DESCRIPTION "Algorithm provides some common helpers."
)
