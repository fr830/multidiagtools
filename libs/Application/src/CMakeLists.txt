
# config.h is generated in the root of the build tree
include_directories("${CMAKE_BINARY_DIR}")

# set(CMAKE_INCLUDE_CURRENT_DIR ON)

find_package(Qt5
  COMPONENTS Widgets
)

set(SOURCE_FILES
  Mdt/StandardPaths.cpp
  Mdt/Application.cpp
)

mdt_add_library(
  NAME Application
  SOURCE_FILES ${SOURCE_FILES}
  HEADERS_DIRECTORY .
  LINK_DEPENDENCIES QtSingleApplication Error
)
mdt_set_library_description(
  NAME Application
  DESCRIPTION "Helper to create main (init Error subsystem, load translations, etc..)"
)
