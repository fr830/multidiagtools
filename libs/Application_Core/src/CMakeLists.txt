
find_package(Qt5 COMPONENTS Core)

configure_file(
  Mdt/Version.h.in
  Version.h
)
set(CMAKE_INCLUDE_CURRENT_DIR ON)

set(SOURCE_FILES
  Mdt/StandardPaths.cpp
  Mdt/CoreApplicationImpl.cpp
  Mdt/CoreApplication.cpp
  Mdt/AbstractConsoleApplicationMainFunction.cpp
)

mdt_add_library(
  NAME Application_Core
  SOURCE_FILES ${SOURCE_FILES}
  HEADERS_DIRECTORY .
  LINK_DEPENDENCIES Error_Core Qt5::Core
)
target_compile_definitions(Application_Core PRIVATE QT_NO_CAST_FROM_ASCII QT_NO_CAST_TO_ASCII QT_NO_CAST_FROM_BYTEARRAY)

mdt_set_library_description(
  NAME Application_Core
  DESCRIPTION "Helper to create main (init Error subsystem, load translations, etc..)"
)
