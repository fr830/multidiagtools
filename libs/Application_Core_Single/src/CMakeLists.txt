
find_package(Qt5 COMPONENTS Core)

# set(CMAKE_INCLUDE_CURRENT_DIR ON)

set(SOURCE_FILES
  Mdt/SingleCoreApplication.cpp
)

mdt_add_library(
  NAME Application_Core_Single
  SOURCE_FILES ${SOURCE_FILES}
  HEADERS_DIRECTORY .
  LINK_DEPENDENCIES Application_Core QtSingleApplication_Core Qt5::Core
)
target_compile_definitions(Application_Core_Single PRIVATE QT_NO_CAST_FROM_ASCII QT_NO_CAST_TO_ASCII QT_NO_CAST_FROM_BYTEARRAY)

mdt_set_library_description(
  NAME Application_Core_Single
  DESCRIPTION "Helper, using QtSingleCoreApplication, to create a single instance application main"
)
