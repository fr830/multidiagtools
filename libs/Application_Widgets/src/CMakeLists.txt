
find_package(Qt5 COMPONENTS Widgets)

# set(CMAKE_INCLUDE_CURRENT_DIR ON)

set(SOURCE_FILES
  Mdt/ApplicationImpl.cpp
  Mdt/Application.cpp
)

mdt_add_library(
  NAME Application_Widgets
  SOURCE_FILES ${SOURCE_FILES}
  HEADERS_DIRECTORY .
  LINK_DEPENDENCIES Application_Core Qt5::Widgets
)
target_compile_definitions(Application_Widgets PRIVATE QT_NO_CAST_FROM_ASCII QT_NO_CAST_TO_ASCII QT_NO_CAST_FROM_BYTEARRAY)

mdt_set_library_description(
  NAME Application_Widgets
  DESCRIPTION "Helper to create main (init Error subsystem, load translations, etc..)"
)
