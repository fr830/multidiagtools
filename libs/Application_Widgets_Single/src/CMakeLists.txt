
find_package(Qt5 COMPONENTS Widgets)

# set(CMAKE_INCLUDE_CURRENT_DIR ON)

set(SOURCE_FILES
  Mdt/SingleApplication.cpp
)

mdt_add_library(
  NAME Application_Widgets_Single
  SOURCE_FILES ${SOURCE_FILES}
  HEADERS_DIRECTORY .
  LINK_DEPENDENCIES Application_Core QtSingleApplication_Widgets Qt5::Widgets
)
target_compile_definitions(Application_Widgets_Single PRIVATE QT_NO_CAST_FROM_ASCII QT_NO_CAST_TO_ASCII QT_NO_CAST_FROM_BYTEARRAY)

mdt_set_library_description(
  NAME Application_Widgets_Single
  DESCRIPTION "Helper, using QtSingleApplication, to create a single instance application main"
)
