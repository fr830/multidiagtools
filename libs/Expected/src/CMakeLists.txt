
set(SOURCE_FILES
    Mdt/Expected.cpp
)

mdt_add_library(
  NAME Expected
  SOURCE_FILES ${SOURCE_FILES}
  HEADERS_DIRECTORY .
  LINK_DEPENDENCIES Error_Core
)
mdt_set_library_description(
  NAME Expected
  DESCRIPTION "Expected provides a basic support of expected value."
)
